server/task_scheduler.go

```diff
@@ -32,8 +32,12 @@ func NewTaskScheduler(server *Server) *TaskScheduler {
 type TaskScheduler struct {
 	executorGetters  map[api.TaskType]func() TaskExecutor
 	runningExecutors map[int]TaskExecutor
-	taskProgress     sync.Map
-	server           *Server
+	taskProgress     sync.Map // map[taskID]api.Progress
+	// sharedGhost is to share states between gh-ost sync and cutover task.
+	// inserted by sync task.
+	// deleted by cutover task.
+	sharedGhost sync.Map // map[taskID]ghostState
```

Comment:
It's a bit strange to put a specific gh-ost related state in the task scheduler.
How about make it
```go
sharedTaskStates sync.Map // map[taskID]interface{}
```
and let the task executors to use it. Like `Context.Value()`.